x/evm/keeper/bank.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/evmos/evmos/v19/x/evm/types"
+)
+
+// BankWrapper is a wrapper around the Cosmos SDK bank keeper
+// that is used to manage an evm denom with 6 or 18 decimals.
+// The wrapper makes the corresponding conversions to achieve:
+//   - With the EVM, the wrapper works always with 18 decimals.
+//   - With the Cosmos bank module, the wrapper works always
+//     with the bank module decimals (either 6 or 18).
+type BankWrapper struct {
+	types.BankKeeper
+	decimals uint32
+}
+
+// NewBankWrapper creates a new bank Keeper wrapper instance.
+// The BankWrapper is used to manage an evm denom with 6 or 18 decimals
+func NewBankWrapper(
+	bk types.BankKeeper,
+) *BankWrapper {
+	return &BankWrapper{
+		bk,
+		types.DefaultDenomDecimals,
+	}
+}
+
+// WithDecimals function updates the decimals on the bank wrapper
+// This function is useful when updating the evm params (denomDecimals)
+func (w *BankWrapper) WithDecimals(decimals uint32) error {
+	if decimals != types.Denom18Dec && decimals != types.Denom6Dec {
+		return fmt.Errorf("decimals = %d not supported. Valid values are %d and %d", decimals, types.Denom18Dec, types.Denom6Dec)
+	}
```

Comment:
We should extract this logic to be sure that it is always in sync with this: https://github.com/evmos/evmos/blob/e5383a9f9a9bbc0346474c3b6d56f90ecfad4cc0/x/evm/types/params.go#L247-L249

Reply:
I agree - I commented on your review comment above, creating a `DenomDecimals.Validate()` method would be cleanest probably